test/TPP/perf/perf-ops.mlir

```diff
@@ -0,0 +1,107 @@
+// RUN: tpp-opt %s -split-input-file -canonicalize | FileCheck %s
+
+// CHECK-LABEL: @perf_matmul_bench
+func.func @perf_matmul_bench(%A: tensor<4x8xf32>,
+          %B: tensor<8x4xf32>, %C: tensor<4x4xf32>, %n: i64) {
+  // CHECK: perf.bench
+  %deltas = perf.bench (%n) {
+    // CHECK: linalg.matmul
+    %D = linalg.matmul ins(%A, %B: tensor<4x8xf32>, tensor<8x4xf32>) outs(%C: tensor<4x4xf32>) -> tensor<4x4xf32>
+    // CHECK: perf.do_not_opt
+    perf.do_not_opt(%D) : tensor<4x4xf32>
```

Comment:
I thought the `perf.yield` was supposed to be inserted when there's no other yield op?

Reply:
After simple parsing, the yield op stays implicit. I believe it is in line with `scf.for` behavior.
But good point anyway as I have not tested the `BenchOp::build` builder. A custom builder might be needed to cover the yield insertion.

Reply:
Yield insertion was missing, solved with a custom `BenchOp::build`.